catkin_home/src/navigation/base_control/src/base_odom.cpp

```diff
@@ -144,15 +141,15 @@ int main(int argc, char** argv)
     odom.twist.twist.angular.y = 0.0;
     odom.twist.twist.angular.z = vth;
     
-    //set the covariance
-    odom.twist.covariance[0] = 1e3; //0.2;
-    odom.twist.covariance[7] = 1e3; //0.2;
+    // Set the covariance.
+    odom.twist.covariance[0] = 1e3;
+    odom.twist.covariance[7] = 1e3;
     odom.twist.covariance[14] = 1e100;
     odom.twist.covariance[21] = 1e100;
     odom.twist.covariance[28] = 1e100;
-    odom.twist.covariance[35] = 1e3; //0.2;
+    odom.twist.covariance[35] = 1e3;
 
-    //publish the message
+    // Publish the message.
     odom_pub.publish(odom);
```

Comment:
Why do you need to publish /odom topic if you are using sendTransform() to publish it directly to the reference frame?

Reply:
In the topic i am able to see the velocity of the robot, in the transform i am not. This is useful to me right now in testing, maybe it could be removed in the future.

Reply:
encapsulate it with an if, and such if to be enabled or disabled with a flag